homeassistant/components/zha/entities/entity.py

```diff
@@ -91,6 +93,16 @@ def zdo_command(self, tsn, command_id, args):
         """Handle a ZDO command received on this cluster."""
         pass
 
+    @callback
+    def device_announce(self, device):
+        """Handle device_announce zdo event."""
+        self.async_schedule_update_ha_state(force_refresh=True)
+
+    @callback
+    def permit_duration(self, permit_duration):
```

Comment:
Is this overwritten in child classes?

Reply:
no it is a side effect of how zigpy event listeners work. If you don't implement all events that an emitter can emit it throws warnings / errors and pollutes the logs. So, because the ZDO listener can emit this event we have to have a handler for it.